E0205 needs to be updated to new format

From: src/test/compile-fail/E0205.rs

Error E0205 needs a span label and updated title, changing it from: 

```
error[E0205]: the trait `Copy` may not be implemented for this type; variant `Bar` does not implement `Copy`
  --> src/test/compile-fail/E0205.rs:18:10
   |
18 | #[derive(Copy)] //~ ERROR E0205
   |          ^^^^
src/test/compile-fail/E0205.rs:18:10: 18:14 note: in this expansion of #[derive(Copy)] (defined in src/test/compile-fail/E0205.rs)

error[E0205]: the trait `Copy` may not be implemented for this type; variant `Bar` does not implement `Copy`
  --> src/test/compile-fail/E0205.rs:16:1
   |
16 | impl Copy for Foo { } //~ ERROR E0205
   | ^^^^^^^^^^^^^^^^^^^^^
```

to:

```
error[E0205]: the trait `Copy` may not be implemented for this type
  --> src/test/compile-fail/E0205.rs:18:10
   |
18 | #[derive(Copy)] //~ ERROR E0205
   |          ^^^^ variant `Bar` does not implement `Copy`
src/test/compile-fail/E0205.rs:18:10: 18:14 note: in this expansion of #[derive(Copy)] (defined in src/test/compile-fail/E0205.rs)

error[E0205]: the trait `Copy` may not be implemented for this type
  --> src/test/compile-fail/E0205.rs:16:1
   |
16 | impl Copy for Foo { } //~ ERROR E0205
   | ^^^^^^^^^^^^^^^^^^^^^ variant `Bar` does not implement `Copy`
```

Bonus: the span for the second one could be shrunk to just the Copy part:

```
error[E0205]: the trait `Copy` may not be implemented for this type
  --> src/test/compile-fail/E0205.rs:18:10
   |
18 | #[derive(Copy)] //~ ERROR E0205
   |          ^^^^ variant `Bar` does not implement `Copy`
src/test/compile-fail/E0205.rs:18:10: 18:14 note: in this expansion of #[derive(Copy)] (defined in src/test/compile-fail/E0205.rs)

error[E0205]: the trait `Copy` may not be implemented for this type
  --> src/test/compile-fail/E0205.rs:16:1
   |
16 | impl Copy for Foo { } //~ ERROR E0205
   |      ^^^^ variant `Bar` does not implement `Copy`
```

Note: I'm not sure the "expansion of..." part is helpful, you could also remove that.

Seems related to https://github.com/rust-lang/rust/issues/35381

--- a/src/librustc_typeck/coherence/mod.rs
+++ b/src/librustc_typeck/coherence/mod.rs
@@ -321,13 +321,18 @@ impl<'a, 'gcx, 'tcx> CoherenceChecker<'a, 'gcx, 'tcx> {
 
                 }
                 Err(CopyImplementationError::InfrigingVariant(name)) => {
-                       struct_span_err!(tcx.sess, span, E0205,
-                                 "the trait `Copy` may not be \
-                                          implemented for this type")
-                           .span_label(span, &format!("variant \
-                                          `{}` does not implement `Copy`",
-                                         name))
-                           .emit()
+                    let item = tcx.map.expect_item(impl_node_id);
+                    let span = if let ItemImpl(_, _, _, Some(ref tr), _, _) = item.node {
+                        tr.path.span
+                    } else {
+                        span
+                    };
+
+                    struct_span_err!(tcx.sess, span, E0205,
+                                     "the trait `Copy` may not be implemented for this type")
+                        .span_label(span, &format!("variant `{}` does not implement `Copy`",
+                                                   name))
+                        .emit()
                 }
                 Err(CopyImplementationError::NotAnAdt) => {
                        span_err!(tcx.sess, span, E0206,

--- a/src/test/compile-fail/E0205.rs
+++ b/src/test/compile-fail/E0205.rs
@@ -14,11 +14,11 @@ enum Foo {
 }
 
 impl Copy for Foo { }
-//~^ ERROR E0205
+//~^ ERROR the trait `Copy` may not be implemented for this type
 //~| NOTE variant `Bar` does not implement `Copy`
 
 #[derive(Copy)]
-//~^ ERROR E0205
+//~^ ERROR the trait `Copy` may not be implemented for this type
 //~| NOTE variant `Bar` does not implement `Copy`
 //~| NOTE in this expansion of #[derive(Copy)]
 enum Foo2<'a> {

